package-res/resources/web/pentaho/type/_doc/spec/ITypeConfiguration.jsdoc

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The `spec.ITypeConfiguration` interface describes
+ * a list of [configuration rules]{@link pentaho.type.spec.ITypeConfiguration#rules}
+ * used to configure one or more [value types]{@link pentaho.type.Value.Type} of
+ * the Pentaho Client Metadata Model.
+ *
+ * The following example is an hypothetical type configuration,
+ * where the fictitious developer, `"john"`,
+ * states that only he should see the visualization he is currently developing,
+ * `"my/radial/bar"`, and also, only when working in Pentaho Analyzer:
+ *
+ * ```js
+ * var myTypeConfig = {
+ *   rules: [
+ *     // Disable a still experimental Viz.
+ *     {
+ *       select: {
+ *         type: "my/radial/bar"
+ *       },
+ *       apply: {
+ *         isBrowsable: false
+ *       }
+ *     },
+ *
+ *     // Enable it, only for the dev user, "john", when in Analyzer
+ *     {
+ *       select: {
+ *         type:        "my/radial/bar",
+ *         user:        "john",
+ *         application: "pentaho.analyzer"
+ *       },
+ *       apply: {
+ *         isBrowsable: true
+ *       }
+ *     }
+ *   ]
+ * };
+ * ```
+ *
+ * ### Rule Selection
+ *
+ * A rule is selected by a given type and context variables, if, together,
+ * these match the rule's [selection variables]{@link pentaho.type.spec.ITypeConfigurationRule#select}.
```

Comment:
Suggested wording: A rule is selected by a given type and context variables, if the type and variables match the rule's [selection variables]{@link pentaho.type.spec.ITypeConfigurationRule#select}.